### ISSUE TITLE
How to specify case-insensitive regex matches in --exclude_from items?

### ISSUE THREAD

**Issue by jim-collier** [borgbackup/borg#5308]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Question

#### Your borg version (borg -V).

borg 1.1.13

#### Operating system (distribution) and version.

Ubuntu 18.04.4

#### Hardware / network configuration, and filesystems used.

Intel(R) Core(TM)2 Duo CPU E8400 @ 3.00GHz
16 GB RAM
ZFS 5 x 3-way mirrors, 22 TiB usable

#### How much data is handled by borg?

15 TiB

## Question

I'm using an `--exclude_from` file. Perl regular expression syntax. The documentation seems to make no mention of how to specify case-insensitive matches. If I include the regex syntax `/i` for that at the end, it's just going to assume that's part of the intended directory match. So how/where does that get specified, or can it even be?

Also related: In the documentation examples, it is stated that directories should be specified simply with "/" (whether Windows or other). But "/" means something different in pure regex, so should those be escaped? They have to be to test the regex. And if they don't HAVE to be escaped in borg (e.g. borg escapes them itself before applying...or something), will it still work if they ARE escaped? For example: to minimize the chance of errors, I'd rather copy straight from a regex tester, without modification, if possible - which would mean leaving in the escapes for slashes. In other words, rather than, say `re:.*/foo/bar/.*`, it would be `re:.*\/foo\/bar\/.*`. Would that still work?

Many thanks.

**Comment by infectormp** [comment]
The regular expression syntax is described in the Python documentation for the [re module](https://docs.python.org/3/library/re.html#re.I).
try (?i) before regex. Something like (?i)[a-z]

**Comment by jim-collier** [comment]
Thanks for that, yes that seems obvious now! In regex testing, I was mistakenly thinking the syntax was one of many permutations that included `/`. Of course, I tried every one except without that.

What about escaping the slashes? (Ideally it can handle escaped slashes. I'd assume it would, since unnecessarily escaping things doesn't usually seem to be a problem. But if borg is pre-escaping slashes before feeding to regex - which seems probable given that escaped slashes isn't a requirement - then that could lead to trouble. I'm about to just test it myself but it's going to take a while to get to that point.

**Comment by ThomasWaldmann** [comment]
If the case-insensitive regex worked and it not documented yet, a PR against the docs would be welcome as this is a common usecase when dealing with files created on DOS/Windows (and maybe macOS, too).

**Comment by ThomasWaldmann** [comment]
Not sure I understand your slashes question. You only would need to deal specially with `backslashes`, but as there usually are none as borg uses forward slashes for pathes, this is a non-issue.

**Comment by jim-collier** [comment]
@ThomasWaldmann I don't know if the case-insensitivity is documented or not, but since it's standard regex, personally I don't think it would be a big oversight if it wasn't. (Nice, but not a big problem.) Personally, I was aware of the (?i) syntax, it just slipped my mind.

As for slashes, I'm aware that backslashes need to be escaped, but for some reason was unclear on forward slashes. Usually when in doubt, I just escape everything in question. (Since it's nigh impossible for any one person to hold ALL of regex syntax and rules in mind at once - most people actively use a subset.) When I wrote that about slashes I was under the (apparently mistaken) assumption that borg didn't support all of python regex. I've since learned that doesn't seem to be the case. (Supports everything I typically use it for.)

Thanks. I'll hit the close button now.
